### PR TITLE
feat(burner): update init to remove undeployed burners from storage

### LINKED ISSUE #135: [burner]: Delete burners if not deployed
### Feature Request

The Burner manager should delete the burners if they don't exist. This will stop user errors trying to use old burners.

### Proposed Solution

_No response_

### Alternatives

_No response_

### Related Code

_No response_

### Additional context

_No response_

### If the feature is accepted, would you be willing to contribute it?

- [ ] Yes I would be willing to contribute

### PR DESCRIPTION
**Pull request by Matth26**
Closes #135 

## Introduced changes

The init function of the burnerManager now checks the deployment status of burner accounts retrieved from local storage.
If a burner account is found to be undeployed (i.e., the deployment transaction is either not present, failed, or incomplete), the init function removes such an account from storage. This helps in maintaining a cleaner, more accurate list of burner accounts, ensuring only successfully deployed accounts are retained.

## Checklist

-   [x] Linked relevant issue
-   [ ] Updated relevant documentation
-   [x] Added relevant tests
-   [ ] Add a dedicated CI job for new examples
-   [x] Performed self-review of the code

### CODE REVIEW

File: packages/create-burner/src/hooks/useBurner.ts [hunk 1]
```diff
@@ -52,8 +52,12 @@ export const useBurner = () => {
...
     useEffect(() => {
-        burnerManager.init();
-        setAccount(burnerManager.getActiveAccount());
+        const initializeBurnerManager = async () => {
+            await burnerManager.init();
+            setAccount(burnerManager.getActiveAccount());
+        };
+
+        initializeBurnerManager();
     }, []);
```
Comment (ponderingdemocritus) [inline packages/create-burner/src/hooks/useBurner.ts:61]:
```
useEffect(() => {
    (async () => {
        await burnerManager.init();
        setAccount(burnerManager.getActiveAccount());
    })();
}, []);
```

File: packages/create-burner/src/hooks/useBurnerManager.ts [hunk 2]
```diff
@@ -26,8 +26,12 @@ export const useBurnerManager = ({
...
     // On mount, initialize the burner manager and set the active account.
     useEffect(() => {
-        burnerManager.init();
-        setAccount(burnerManager.getActiveAccount());
+        const initializeBurnerManager = async () => {
+            await burnerManager.init();
+            setAccount(burnerManager.getActiveAccount());
+        };
+
+        initializeBurnerManager();
```
Comment (ponderingdemocritus) [inline packages/create-burner/src/hooks/useBurnerManager.ts:34]:
```
useEffect(() => {
    (async () => {
        await burnerManager.init();
        setAccount(burnerManager.getActiveAccount());
    })();
}, []);
```

File: packages/create-burner/src/manager/burnerManager.ts [hunk 3]
```diff
@@ -99,34 +98,42 @@ export class BurnerManager {
...
     }
 
-    public init(): void {
+    private async isBurnerDeployed(deployTx: string): Promise<boolean> {
+        try {
+            const receipt =
+                await this.masterAccount.getTransactionReceipt(deployTx);
+            return receipt !== null;
+        } catch (error) {
+            return false; // Assume not deployed if there's an error fetching the receipt
```
Comment (ponderingdemocritus) [inline packages/create-burner/src/manager/burnerManager.ts:107]:
```suggestion
            return false;
```

File: packages/create-burner/src/manager/burnerManager.ts [hunk 4]
```diff
@@ -99,34 +98,42 @@ export class BurnerManager {
...
+
+    public async init(): Promise<void> {
         const storage = this.getBurnerStorage();
+        const addresses = Object.keys(storage);
+
+        const checks = addresses.map(async (address) => {
+            const isDeployed = await this.isBurnerDeployed(
+                storage[address].deployTx
+            );
+            return isDeployed ? null : address; // Keep address if not deployed, otherwise null
```
Comment (ponderingdemocritus) [inline packages/create-burner/src/manager/burnerManager.ts:119]:
```suggestion
            return isDeployed ? null : address;
```

File: packages/create-burner/src/hooks/useBurner.ts (hunk 1 above)
Comment (Matth26) [reply packages/create-burner/src/hooks/useBurner.ts]: nice syntactic sugar !

**Pull request merged**